e2e/tests/regression/test_backup.robot

```diff
@@ -135,3 +135,19 @@ Test Uninstallation With Backups
     And Check Longhorn CRD removed
 
     Then Install Longhorn
+
+Test Cleanup Snapshot With The Global Setting After Backup Completed
+    [Documentation]    Test cleanup snapshot with the global setting after backup completed
+    Given Create volume 0 with    dataEngine=${DATA_ENGINE}
+    And Attach volume 0
+    And Wait for volume 0 healthy
+
+    When Write data 0 to volume 0
+    And Create backup 0 for volume 0
+    And Check snapshot for backup 0 of volume 0 exists True
+
+    When Set setting auto-cleanup-after-on-demand-backup-completed to true
+    And Write data 1 to volume 0
+    And Create backup 1 for volume 0
+    And Check snapshot for backup 1 of volume 0 exists False
```

Comment:
Will `snapshot for backup 0 of volume 0` also be removed after enabling `auto-cleanup-after-on-demand-backup-completed`?

From the test step 8 of https://github.com/longhorn/longhorn/issues/9213#issuecomment-2609099226, does it mean the backup created by a recurring job will also be affected by this setting?

Will a snapshot created by a `system backup` also be removed if this setting is enabled?

Reply:
>Will snapshot for backup 0 of volume 0 also be removed after enabling auto-cleanup-after-on-demand-backup-completed?

No, only on-demand backups created after the setting is enabled will be affected.

>From the test step 8 of https://github.com/longhorn/longhorn/issues/9213#issuecomment-2609099226, does it mean the backup created by a recurring job will also be affected by this setting?

Updated step 8 of https://github.com/longhorn/longhorn/issues/9213#issuecomment-2609099226, because this setting will not affect the recurring job behavior.

>Will a snapshot created by a system backup also be removed if this setting is enabled?

No, only delete the snapshot CR after the on-demand backup is completed.